File: problems/621_task_scheduler/solution.py
class Solution:
    def leastInterval(self, tasks: list[str], n: int) -> int:
        from collections import Counter
        freq = list(Counter(tasks).values())
        max_freq = max(freq)
        max_count = freq.count(max_freq)
        return max(len(tasks), (max_freq - 1) * (n + 1) + max_count)

File: problems/621_task_scheduler/Solution.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] freq = new int[26];
        for (char t : tasks) freq[t - 'A']++;
        int maxFreq = 0, maxCount = 0;
        for (int f : freq) {
            if (f > maxFreq) { maxFreq = f; maxCount = 1; }
            else if (f == maxFreq) maxCount++;
        }
        return Math.max(tasks.length, (maxFreq - 1) * (n + 1) + maxCount);
    }
}
